### PR TITLE
Move now-working boxed dyn trait test

### LINKED ISSUE #61: Not implemented: gotoc-dynamic-trait-box
Dynamic trait inside a box. Test implemented: gotoc-dynamic-trait-box but currently failing/not parsing through rustc.

### PR DESCRIPTION
**Pull request by avanhatt**
### Description of changes: 

#61 tracked a previously-unimplemented boxed dynamic trait behavior, but this now works (since at least e105c6b4784). So move the test to our standard regression tests.

### Resolved issues:

Resolves #61

### Testing:

* How is this change tested?
Existing tests.
* Is this a refactor change?
No.

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by bdalrhm — APPROVED** [review]
Not sure if my review counts, but LGTM!

**Comment by adpaco-aws** [comment]
Just realized that this was a `run-make` test which is being moved to the `cbmc` tests. Usually, `run-make` tests have an associated `expected` file (being removed here) containing a variable number of lines which are sub-strings of CBMC's output for that test. See `src/test/run-make/gotoc-vecdq/expected` for an example.

Would this test add value to our `cbmc` tests for dynamic traits? If not, then I would edit the `expected` file with lines appearing in CBMC's output and leave the test in there.

File: src/test/cbmc/DynTrait/boxed-trait.rs [hunk 1]
```diff
@@ -51,5 +51,5 @@ fn main() {
 
     let square = Box::new(Square { w: 3 });
     assert!(square.vol(3) == 27);
-    //   assert!(do_vol(&square, 2) == 18);
+    assert!(do_vol(&*square, 2) == 18);
```
Comment (danielsn) [inline src/test/cbmc/DynTrait/boxed-trait.rs:54]: we should probably add a `_fail` test too

**Review by danielsn — CHANGES_REQUESTED** [review]
Add a _fail test

**Comment by avanhatt** [comment]
> Just realized that this was a `run-make` test which is being moved to the `cbmc` tests. Usually, `run-make` tests have an associated `expected` file (being removed here) containing a variable number of lines which are sub-strings of CBMC's output for that test. See `src/test/run-make/gotoc-vecdq/expected` for an example.
> 
> Would this test add value to our `cbmc` tests for dynamic traits? If not, then I would edit the `expected` file with lines appearing in CBMC's output and leave the test in there.

Because this test now works as expected, it seems to fit better into the `cbmc` tests, since those are executed on every PR. The pattern of calling code with `do_vol(&*rec, 2)` is slightly (though not significantly) different from the other `DynTrait` tests. It seems like it would be better to just delete this test than keep it in `make-run`, since the output is no longer noteworthy?

**Comment by adpaco-aws** [comment]
Yes, this test was not being executed before. I think only `run-make` tests which are located in `gotoc*` folders are. It seems to me that moving this test to `cbmc` is the right thing to do.

**Review by danielsn — APPROVED** [review]

**Pull request merged**